modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/JSON/TokenGroupIdentifier.swift

```diff
@@ -162,6 +168,10 @@ public class TokenGroupIdentifier: TokenGroupIdentifierProtocol {
     public func hasContract(address: String, chainID: Int) -> Bool {
         return decodedTokenEntries["\(address):\(chainID)"] != nil
     }
+
+    public func isSpam(address: String, chainID: Int) -> Bool {
```

Comment:
I wondered if the function signature should be this instead:

```
func isSpam(address: AlphaWallet.Address, chain: RPCServer) -> Bool
```

On one hand, this is stronger typing (and we can change the implementation so the key is a new type instead of appending lowercased strings).

On the other hand, this introduces a dependency on 2 pods into `TokenGroupIdentifier.swift`. The dependency is not an issue now, but if/when we extract it into a pod, it would be. For example when we extracted the `ENS` pod https://github.com/AlphaWallet/alpha-wallet-ios/blob/eefc84d63bd800a4c6690a3750d4f0dcaede3a6d/modules/AlphaWalletENS/AlphaWalletENS/ENS.swift#L22, we replace RPCServer with a `ChainId = Int` typealias.

I guess we can leave it for now. Doesn't matter in the grand scheme of things and is easy to change